lib/localparticipant.js

```diff
@@ -259,13 +259,19 @@ LocalParticipant.prototype._getOrCreateLocalTrackPublication = function _getOrCr
 LocalParticipant.prototype.addTrack = function addTrack(track) {
   this._log.deprecated('LocalParticipant#addTrack has been deprecated. '
     + 'Use LocalParticipant#publishTrack instead.');
-  return this._addTrack(util.asLocalTrack(track, {
-    log: this._log,
+  util.validateLocalTrack(track, {
     LocalAudioTrack: this._LocalAudioTrack,
     LocalDataTrack: this._LocalDataTrack,
     LocalVideoTrack: this._LocalVideoTrack,
     MediaStreamTrack: this._MediaStreamTrack
-  }));
+  });
+  if (this.tracks.has(track.id)) {
+    return null;
+  }
+  this.publishTrack(track).catch(function() {
+    // Do nothing.
+  });
+  return track;
```

Comment:
@markandrus I think this should be: `return this.tracks.get(track.id);` because, `track` can also be a `MediaStreamTrack`.

Reply:
Ah, very good point 👍